lib/create-http-client.js

```diff
@@ -6,7 +6,7 @@ import { isNode, getNodeVersion } from './utils'
 
 // Matches 'sub.host:port' and extracts hostname and port
 // Also enforces toplevel domain specified, no spaces and no protocol
-const HOST_REGEX = /^(?!\w+:\/\/)([^\s:]+\.[^\s:]+)(?::(\d+))?(?!:)$/
+const HOST_REGEX = /^(?!\w+:\/\/)([^\s:]+\.?[^\s:]+)(?::(\d+))?(?!:)$/
```

Comment:
What does this match optionally? My regex not good.

Reply:
Does the comment above still apply based the new regex?

Reply:
The period should be optional (when it's mandatory, you can't set your host = `localhost`, for example). I tested it with an online regex checker, but we could also surround the period with [] if that's clearer?

Reply:
my regex are also not 💯

Reply:
also good point-- I'll modify the comment.